### PR TITLE
[Dynamic buffer] Fix enable-dynamic-buffer.py issue

### PR DESCRIPTION
**Pull request by JibinBao**
### Description of PR

Before stopping buffermgrd, need to make sure buffermgrd is running, Otherwise it might cause some side-effect timing issue.  For example: after loading minigraph, buffermgrd will restart, if we run command to stop buffermgrd when buffermgrd is not running, the buffermgrd will not be stopped really, and the consequence is the command starting buffermgrd triggered by loading minigraph in the queue will be executed in the later. Therefore, when the subsequent starting buffermgrd happens between time changing the config db with the dynamic buffer config and setting the buffer model to dynamic, it will raise some error by sai, because dynamic buffer config will be applied to traditional model.

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [x] 202405

#### What is the motivation for this PR?
Fix enable-dynamic-buffer issue

#### How did you do it?
Before stopping buffermgrd, check if it is running

#### How did you verify/test it?
runing it on mellanox device

#### Any platform specific information?
Mellnax device

### CODE REVIEW

**Review by stephenxs — APPROVED** [review]

**Comment by bingwang-ms** [comment]
@kperumalbfn Can you help review?

File: tests/common/helpers/enable-dynamic-buffer.py [hunk 1]
```diff
@@ -131,6 +136,28 @@ def stop_traditional_buffer_model(config_db):
...
+def check_buffermgrd_is_running():
+    cmd_get_buffermgrd_status = "docker exec swss supervisorctl status buffermgrd"
+    max_try_times = 10
+    try_times = 0
+    while try_times < max_try_times:
+        try_times += 1
+        proc = subprocess.Popen(cmd_get_buffermgrd_status, shell=True, stdout=subprocess.PIPE)
+        output, err = proc.communicate()
+        if err:
+            logger.log_notice("Failed to check buffermgrd status: {}".format(err))
```
Comment (kperumalbfn) [inline tests/common/helpers/enable-dynamic-buffer.py:148]: Does this require retry? If so, please add retry_timer to the log like below log message.

File: tests/common/helpers/enable-dynamic-buffer.py [hunk 2]
```diff
@@ -119,6 +120,10 @@ def stop_traditional_buffer_model(config_db):
     # Stop the buffermgrd
     # We don't stop the buffermgrd at the beginning
     # because we need it to remove tables from APPL_DB while their counter part are removed from CONFIG_DB
+
+    # Before stopping buffermgrd, need to make sure buffermgrd is running,
+    # otherwise it might cause some side-effect timing issue
+    check_buffermgrd_is_running()
```
Comment (kperumalbfn) [inline tests/common/helpers/enable-dynamic-buffer.py:126]: Check the return status before stopping buffermgrd?

Comment (JibinBao) [reply tests/common/helpers/enable-dynamic-buffer.py:126]: Yes

**Comment by JibinBao** [comment]
Hi @kperumalbfn,
Comments have been fixed.  Can you help review it again?

**Review by kperumalbfn — APPROVED** [review]

**Comment by JibinBao** [comment]
Hi @bingwang-ms  Can you help approve it?

**Pull request merged**